packages/squiggle-lang/__tests__/Distributions/AlgebraicCombination_test.res

```diff
@@ -0,0 +1,313 @@
+
+    describe("mean", () => {
+        test("normal(mean=5) + normal(mean=20)", () => {
+            normalDist5
+            -> algebraicAdd(normalDist20)
+            -> E.R2.fmap(GenericDist_Types.Constructors.UsingDists.mean)
+            -> E.R2.fmap(run)
+            -> E.R2.fmap(toFloat)
+            -> E.R.toExn
+            -> expect
+            -> toBe(Some(2.5e1))
+        })
+
+        test("uniform(low=9, high=10) + beta(alpha=2, beta=5)", () => {
+            // let uniformMean = (9.0 +. 10.0) /. 2.0
+            // let betaMean = 1.0 /. (1.0 +. 5.0 /. 2.0)
+            let received = uniformDist 
+                -> algebraicAdd(betaDist)
+                -> E.R2.fmap(GenericDist_Types.Constructors.UsingDists.mean)
+                -> E.R2.fmap(run)
+                -> E.R2.fmap(toFloat)
+                -> E.R.toExn
+            switch received {
+                | None => "algebraicAdd has" -> expect -> toBe("failed")
+                // This is nondeterministic, we could be in a situation where ci fails but you click rerun and it passes, which is bad. 
+                // sometimes it works with ~digits=2. 
+                | Some(x) => x -> expect -> toBeSoCloseTo(0.01927225696028752, ~digits=1)  // (uniformMean +. betaMean)
+            }
+        })
+        test("beta(alpha=2, beta=5) + uniform(low=9, high=10)", () => {
+            // let uniformMean = (9.0 +. 10.0) /. 2.0
+            // let betaMean = 1.0 /. (1.0 +. 5.0 /. 2.0)
+            let received = betaDist 
+                -> algebraicAdd(uniformDist)
+                -> E.R2.fmap(GenericDist_Types.Constructors.UsingDists.mean)
+                -> E.R2.fmap(run)
+                -> E.R2.fmap(toFloat)
+                -> E.R.toExn
+            switch received {
+                | None => "algebraicAdd has" -> expect -> toBe("failed")
+                // This is nondeterministic, we could be in a situation where ci fails but you click rerun and it passes, which is bad. 
+                // sometimes it works with ~digits=2. 
+                | Some(x) => x -> expect -> toBeSoCloseTo(0.019275414920485248, ~digits=1)  // (uniformMean +. betaMean)
+            }
+        })
+    })
+    describe("pdf", () => {
+        testAll("(normal(mean=5) + normal(mean=5)).pdf (imprecise)", list{8e0, 1e1, 1.2e1, 1.4e1}, x => {
+            let received = normalDist10
+                -> Ok 
+                -> E.R2.fmap(d => GenericDist_Types.Constructors.UsingDists.pdf(d, x))
+                -> E.R2.fmap(run)
+                -> E.R2.fmap(toFloat)
+                -> E.R.toOption
+                -> E.O.flatten
+            let calculated = normalDist5
+                -> algebraicAdd(normalDist5)
+                -> E.R2.fmap(d => GenericDist_Types.Constructors.UsingDists.pdf(d, x))
+                -> E.R2.fmap(run)
+                -> E.R2.fmap(toFloat)
+                -> E.R.toOption
+                -> E.O.flatten
```

Comment:
This pattern of E.R.toOption -> E.O.flatten strikes me as suspicious. I might play with this a bit manually.

Reply:
This is what I landed on after a long journey of looking for ways not to pattern match for the final `expect` statement (playing with `E.O.default`). But if I'm _going_ to be pattern matching the final `expect` anyway, it's probably better to leave it in a `result`.